### PR TITLE
chore(deps): Bump `@metamask/ppom-validator` from `0.32.0` to `0.35.0`

### LINKED ISSUE #11134: chore(deps): Bump `@metamask/ppom-validator` from `0.32.0` to `0.35.1`
### What is this about?

The `PPOMController` is behind by two major versions and needs to be updated.

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by MajorLift**
## **Description**

```md
### Changed

- **BREAKING:** Bump `@metamask/ppom-validator` from `0.32.0` to `0.34.0` ([#11114](https://github.com/MetaMask/metamask-mobile/pull/11114))
```

To fully align the dependencies, a new release of `ppom-validator` will need to be released with version bumps for `@metamask/{base-controller,controller-utils,network-controller,utils}`.

## **Related issues**

- Closes https://github.com/MetaMask/metamask-mobile/issues/11134

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -167,7 +167,7 @@
     "@metamask/permission-controller": "^11.0.0",
     "@metamask/phishing-controller": "^12.0.3",
     "@metamask/post-message-stream": "^8.0.0",
-    "@metamask/ppom-validator": "0.32.0",
+    "@metamask/ppom-validator": "0.35.0",
```
Comment (cryptodev-2s) [inline package.json:170]: we can use `0.35.1` as it has a fix for an issue we discovered and fixed in extension that can block malicious transactions detection

**Comment by mikesposito** [comment]
We currently have four different `@metamask/network-controller` major versions in the dependency tree:

v21
```
=> Found "@metamask/network-controller@21.0.1"
info Has been hoisted to "@metamask/network-controller"
info This module exists because it's specified in "dependencies".
info Disk size without dependencies: "5.07MB"
info Disk size with unique dependencies: "17.91MB"
info Disk size with transitive dependencies: "39.14MB"
info Number of shared dependencies: 60
```

v20
```
=> Found "@metamask/smart-transactions-controller#@metamask/network-controller@20.1.0"
info This module exists because "@metamask#smart-transactions-controller" depends on it.
info Disk size without dependencies: "3.17MB"
info Disk size with unique dependencies: "15.18MB"
info Disk size with transitive dependencies: "36.98MB"
info Number of shared dependencies: 59
```

v19
```
=> Found "@metamask/polling-controller#@metamask/network-controller@19.0.0"
info This module exists because "@metamask#polling-controller" depends on it.
info Disk size without dependencies: "676KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "33.69MB"
info Number of shared dependencies: 58

=> Found "@metamask/assets-controllers#@metamask/network-controller@19.0.0"
info This module exists because "@metamask#assets-controllers" depends on it.
info Disk size without dependencies: "676KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "33.69MB"
info Number of shared dependencies: 58

=> Found "@metamask/gas-fee-controller#@metamask/network-controller@19.0.0"
info This module exists because "@metamask#gas-fee-controller" depends on it.
info Disk size without dependencies: "676KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "33.69MB"
info Number of shared dependencies: 58

=> Found "@metamask/selected-network-controller#@metamask/network-controller@19.0.0"
info This module exists because "@metamask#selected-network-controller" depends on it.
info Disk size without dependencies: "676KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "33.69MB"
info Number of shared dependencies: 58

=> Found "@metamask/smart-transactions-controller#@metamask/transaction-controller#@metamask/network-controller@19.0.0"
info This module exists because "@metamask#smart-transactions-controller#@metamask#transaction-controller" depends on it.
info Disk size without dependencies: "676KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "33.69MB"
info Number of shared dependencies: 58
```

v17
```
=> Found "@metamask/swaps-controller#@metamask/network-controller@17.2.1"
info Reasons this module exists
   - "@metamask#swaps-controller#@metamask#gas-fee-controller" depends on it
   - Hoisted from "@metamask#swaps-controller#@metamask#gas-fee-controller#@metamask#network-controller"
   - Hoisted from "@metamask#swaps-controller#@metamask#gas-fee-controller#@metamask#polling-controller#@metamask#network-controller"
info Disk size without dependencies: "452KB"
info Disk size with unique dependencies: "11.67MB"
info Disk size with transitive dependencies: "35.82MB"
info Number of shared dependencies: 59
```

This version mismatch is visible through TS errors, but it is hard to say if some functionality is broken as well, and very hard to tell by looking at Changelogs since a lot of things changed between these NetworkController major versions

**Comment by cryptodev-2s** [comment]
@mikesposito can we replace this PR with https://github.com/MetaMask/metamask-mobile/pull/12225 ? which is now all green.

**Comment by mikesposito** [comment]
@cryptodev-2s definitely, thanks for pointing that out!

I'm closing this one.

**Pull request closed without merging**